projects/packages/masterbar/tools/webpack.config.js

```diff
@@ -0,0 +1,135 @@
+const path = require( 'path' );
+const jetpackWebpackConfig = require( '@automattic/jetpack-webpack-config/webpack' );
+const RemoveAssetWebpackPlugin = require( '@automattic/remove-asset-webpack-plugin' );
+const glob = require( 'glob' );
+
+const sharedWebpackConfig = {
+	mode: jetpackWebpackConfig.mode,
+	devtool: jetpackWebpackConfig.devtool,
+	output: {
+		...jetpackWebpackConfig.output,
+		path: path.join( __dirname, '../build' ),
+	},
+	optimization: {
+		...jetpackWebpackConfig.optimization,
+	},
+	resolve: {
+		...jetpackWebpackConfig.resolve,
+		modules: [ 'node_modules' ],
+		alias: {
+			...jetpackWebpackConfig.resolve.alias,
+			fs: false,
+		},
+	},
+	node: {},
+	externals: {
+		...jetpackWebpackConfig.externals,
+		jetpackConfig: JSON.stringify( {
+			consumer_slug: 'jetpack-masterbar',
+		} ),
+	},
+	module: {
+		strictExportPresence: true,
+		rules: [
+			// Transpile JavaScript
+			jetpackWebpackConfig.TranspileRule( {
+				exclude: /node_modules\//,
+			} ),
+
+			// Handle CSS.
+			jetpackWebpackConfig.CssRule( {
+				extensions: [ 'css', 'sass', 'scss' ],
+				extraLoaders: [
+					{
+						loader: 'postcss-loader',
+						options: {
+							postcssOptions: { plugins: [ require( 'autoprefixer' ) ] },
+						},
+					},
+					{
+						loader: 'sass-loader',
+						options: {
+							sassOptions: {
+								// The minifier will minify if necessary.
+								outputStyle: 'expanded',
+							},
+						},
+					},
+				],
+			} ),
+
+			// Leave fonts and images in place.
+			{
+				test: /\.(eot|ttf|woff|png|svg)$/i,
+				type: 'asset/resource',
+				generator: {
+					emit: false,
+					filename: '[file]',
+				},
+			},
+		],
+	},
+	plugins: [
+		...jetpackWebpackConfig.StandardPlugins( {
+			DependencyExtractionPlugin: false,
+			I18nLoaderPlugin: false,
+			I18nCheckPlugin: false,
+			MiniCssWithRtlPlugin: false,
+			WebpackRtlPlugin: false,
```

Comment:
This webpack config should mostly work here as things are, because none of the JS files use i18n or modules and you're adding back the `MiniCssWithRtlPlugin` and `WebpackRtlPlugin` plugins where it really matters below. The main thing you'll be missing is that there won't be any `.asset.php` files for the Assets package to get version hashes from.

To do it more right, you'd want to update the JS files to `import` their corresponding CSS files, and any other modules they might need (looks like none need any at the moment). That should let you drop the separate `masterBarCssFiles` and enable all these plugins for the `masterBarJsFiles` block.

One possible drawback to doing that is that a JS file has to exist to `import` the CSS file, even if the JS file is otherwise empty. The Assets package doesn't (currently) have support for CSS-only assets.

Reply:
I gave this a shot with https://github.com/Automattic/jetpack/pull/37342/commits/0fc9c9e9997c710874229eab4728cd938636ddcd